### PR TITLE
Add debug logging of options

### PR DESCRIPTION
**Pull request by bstpierr**
This is for #1768.

### CODE REVIEW

File: Firebase/Core/FIROptions.m [hunk 1]
```diff
@@ -108,6 +108,17 @@ + (FIROptions *)defaultOptions {
 
 #pragma mark - Private class methods
 
+#ifdef DEBUG
++ (void)logOptionsInDictionary:(NSDictionary *)dictionary {
+  if (FIRIsLoggableLevel(FIRLoggerLevelDebug, kFIRLoggerCore)) {
+    [dictionary enumerateKeysAndObjectsUsingBlock:^(id _Nonnull key, id _Nonnull value,
+                                                    BOOL *_Nonnull stop) {
+      FIRLogDebug(kFIRLoggerCore, @"I-COR0000##", @"Read option '%@'='%@'", key, value);
```
Comment (paulb777) [inline Firebase/Core/FIROptions.m]: Why "##"?

**Comment by paulb777** [comment]
The travis unit test failures should be fixed:
```

    ✓ testSystemLogWithDefaultStatus (0.000 seconds)
FIROptionsTest
    ✗ testAnalyticsCollectionExplicitlySet, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testAnalyticsCollectionGlobalSwitchDisabled, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testAnalyticsCollectionGlobalSwitchEnabled, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testAnalyticsCollectionGlobalSwitchOverrideToDisable, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testAnalyticsCollectionGlobalSwitchOverrideToEnable, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testAnalyticsConstants, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testAnalyticsOptions, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testAnalyticsOptions_combinatorial, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testCopyingProperties, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testCopyWithZone, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testDefaultOptions, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testDefaultOptionsDictionaryWithInvalidSourceFile, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testDefaultOptionsDictionaryWithNilFilePath, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testInit, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testInitCustomizedOptions, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testInitWithContentsOfFile, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testVersionConsistency, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
    ✗ testVersionFormat, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
FIROptionsTest
  testAnalyticsCollectionExplicitlySet, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testAnalyticsCollectionGlobalSwitchDisabled, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testAnalyticsCollectionGlobalSwitchEnabled, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testAnalyticsCollectionGlobalSwitchOverrideToDisable, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testAnalyticsCollectionGlobalSwitchOverrideToEnable, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testAnalyticsConstants, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testAnalyticsOptions, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testAnalyticsOptions_combinatorial, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testCopyingProperties, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
  ```
  testCopyWithZone, failed: caught "NSInternalInconsistencyException", "Incorrect message code format."
  /Users/travis/build/firebase/firebase-ios-sdk/Example/Core/Tests/FIROptionsTest.m:42
  ```
  [super setUp];
  [FIROptions resetDefaultOptions];
}
```

**Comment by ryanwilson** [comment]
My bad on this one, looks like I didn't provide the best description in #1768. The original intention was for the property `BOOL isDataCollectionDefaultEnabled` https://github.com/firebase/firebase-ios-sdk/blob/7b43c614aee379d3d5a797e69561a6e4a07aa014/Firebase/Core/FIRApp.m#L384 and the setter, `setDataCollectionDefaultEnabled:` https://github.com/firebase/firebase-ios-sdk/blob/7b43c614aee379d3d5a797e69561a6e4a07aa014/Firebase/Core/FIRApp.m#L361

I do worry about logging `FIROptions` to the console since sharing logs would then include API keys and the `googleAppID` which shouldn't be shared. Apologies for the confusion and the extra work done.

**Comment by bstpierr** [comment]
Oops. I can certainly get `BOOL isDataCollectionDefaultEnabled` logged rather than all of these FIROptions.

You make a good point about sharing logs. I'll drop this PR and whip up another with the correct logging being done (as well as with actual error codes in the logs, as @paulb777 mentioned.).

**Pull request closed without merging**